### PR TITLE
remove unnecessary tests and update test

### PR DESCRIPTION
**Pull request by vankichi**
### Description:

I removed unnecessary test files from current codes.
These files are generated automatically but target mock implementation or type alias of external dependencies.
also, I updated the Makefile to prevent generating such files from now on.
finally, I ran `make gotests/gen` and updated the current test files.

### Environment:

- Go Version: 1.16
- Docker Version: 19.03.8
- Kubernetes Version: 1.18.2
- NGT Version: 1.12.3

### Types of changes:

- [ ] Bug fix [type/bug]
- [ ] New feature [type/feature]
- [x] Add tests [type/test]
- [ ] Security related changes [type/security]
- [ ] Add documents [type/documentation]
- [ ] Refactoring [type/refactoring]
- [ ] Update dependencies [type/dependency]
- [ ] Update benchmarks and performances [type/bench]
- [ ] Update CI [type/ci]

### Changes to Core Features:

- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your core changes, as applicable?
- [ ] Have you successfully ran tests with your changes locally?

### Checklist:

- [x] I have read the [CONTRIBUTING](https://github.com/vdaas/vald/blob/master/CONTRIBUTING.md) document.
- [x] I have checked open [Pull Requests](https://github.com/vdaas/vald/pulls) for the similar feature or fixes?
- [ ] I have added tests and benchmarks to cover my changes.
- [x] I have ensured all new and existing tests passed.
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Review by hlts2 — DISMISSED** [review]
thank you
LGTM

File: Makefile [hunk 1]
```diff
@@ -168,6 +168,12 @@ GO_SOURCES = $(eval GO_SOURCES := $(shell find \
 		./pkg \
 		-not -path './cmd/cli/*' \
 		-not -path './internal/core/algorithm/ngt/*' \
+		-not -path './internal/compress/gob/*' \
```
Comment (kevindiu) [inline Makefile:171]: can you add some comments on the Makefile about why we ignore these packages?

**Review by kevindiu — COMMENTED** [review]
I think this PR requires author review. What do you think?

File: Makefile [hunk 2]
```diff
@@ -161,13 +161,30 @@ PROTO_PATHS = \
 	$(GOPATH)/src/github.com/googleapis/googleapis \
 	$(GOPATH)/src/github.com/gogo/googleapis
 
+# [Warning]
+# The below directories have the no original implementation.
+# You should not add any features.
+# - internal/copress/gob
+# - internal/compress/gzip
+# - internal/compress/lz4
+# - internal/compress/zst
```
Comment (kevindiu) [inline Makefile:170]:
```suggestion
# The below packages have no original implementation.
# You should not add any features.
# - internal/copress/gob
# - internal/compress/gzip
# - internal/compress/lz4
# - internal/compress/zstd
```

Comment (vankichi) [reply Makefile:170]: I wonder which is the correct using 'directories' or 'packages' 🤔

Comment (kevindiu) [reply Makefile:170]: it is source code path so i think it is safe to use the wording `package` 🤔
what do you think?

Comment (vankichi) [reply Makefile:170]: OK

**Review by hlts2 — DISMISSED** [review]
Thank you LGTM

**Review by kevindiu — APPROVED** [review]
LGTM

File: internal/db/rdb/mysql/mysql_test.go [hunk 3]
```diff
@@ -4222,3 +4222,169 @@ func Test_mySQLClient_RemoveIPs(t *testing.T) {
...
+				tlsConfig:            test.fields.tlsConfig,
+				maxOpenConns:         test.fields.maxOpenConns,
+				maxIdleConns:         test.fields.maxIdleConns,
+				session:              test.fields.session,
+				connected:            test.fields.connected,
+				eventReceiver:        test.fields.eventReceiver,
+				dbr:                  test.fields.dbr,
+			}
+
+			m.outputLog(test.args.err)
```
Comment (kevindiu) [inline internal/db/rdb/mysql/mysql_test.go:4384]:
```suggestion
			m. errorLog(test.args.err)
```

File: internal/db/rdb/mysql/mysql_test.go [hunk 4]
```diff
@@ -4222,3 +4222,169 @@ func Test_mySQLClient_RemoveIPs(t *testing.T) {
 		})
 	}
 }
+
+func Test_mySQLClient_outputLog(t *testing.T) {
```
Comment (kevindiu) [inline internal/db/rdb/mysql/mysql_test.go:4226]:
```suggestion
func Test_mySQLClient_errorLog(t *testing.T) {
```

**Review by kevindiu — CHANGES_REQUESTED** [review]
please fix

File: internal/db/rdb/mysql/mysql_test.go (hunk 3 above)
Comment (vankichi) [reply internal/db/rdb/mysql/mysql_test.go:4384]:
```suggestion
			m.errorLog(test.args.err)
```

**Review by hlts2 — APPROVED** [review]
LGTM

**Review by kevindiu — DISMISSED** [review]
LGTM

**Comment by kevindiu** [comment]
Please also remove this file.

internal/compress/mock_test.go

**Review by kevindiu — DISMISSED** [review]
LGTM

**Pull request merged**